doc/code_test.go

```diff
@@ -0,0 +1,84 @@
+package doc
+import (
+	"testing"
+	"go/token"
+	"go/parser"
+	"go/doc"
+)
+
+func TestPrintExampleConstAnnotation(t *testing.T) {
```

Comment:
these tests cases could be in a slice of strings, and you could just have a single test that tests each of them.